### PR TITLE
fix: added missing mime type

### LINKED ISSUE #519: Missing MIME type
## Describe your Issue
Using the default nginx config should include all nessesary mime types.

### Logs and Errors

Currently when i rollout the latest version of nextcloud fpm with nginx sidecar enabled it results in the warning:

Your webserver does not serve `.mjs` files using the JavaScript MIME type. This will break some apps by preventing browsers from executing the JavaScript files. You should configure your webserver to serve `.mjs` files with either the `text/javascript` or `application/javascript` MIME type.

## Describe your Environment

- Kubernetes distribution: kops on hetzner
- Helm Version (or App that manages helm): ArgoCD v2.9.5+f943664
- Helm Chart Version: 4.5.12

- `values.yaml`:

```yaml
nextcloud:
  image:
    repository: nextcloud
    tag: 28.0.2-fpm
    flavor: fpm
    pullPolicy: IfNotPresent
  replicaCount: 1


  ingress:
    enabled: true
    className: nginx
    annotations:
      external.dns/provider: cf
      nginx.ingress.kubernetes.io/proxy-body-size: 4G
      nginx.ingress.kubernetes.io/server-snippet: |-
        server_tokens off;
        proxy_hide_header X-Powered-By;
        rewrite ^/.well-known/webfinger /index.php/.well-known/webfinger last;
        rewrite ^/.well-known/nodeinfo /index.php/.well-known/nodeinfo last;
        rewrite ^/.well-known/host-meta /public.php?service=host-meta last;
        rewrite ^/.well-known/host-meta.json /public.php?service=host-meta-json;
        location = /.well-known/carddav {
          return 301 $scheme://$host/remote.php/dav;
        }
        location = /.well-known/caldav {
          return 301 $scheme://$host/remote.php/dav;
        }
        location = /robots.txt {
          allow all;
          log_not_found off;
          access_log off;
        }
        location ~ ^/(?:build|tests|config|lib|3rdparty|templates|data)/ {
          deny all;
        }
        location ~ ^/(?:autotest|occ|issue|indie|db_|console) {
          deny all;
        }
    tls:
      - hosts:
          - REDACTED
    labels: {}
    path: /
    pathType: Prefix


  phpClientHttpsFix:
    enabled: false
    protocol: https

  nextcloud:
    host: REDACTED
    username: REDACTED
    password: REDACTED
    existingSecret:
      enabled: true
      secretName: nextcloud
      usernameKey: nextcloud-username
      passwordKey: nextcloud-password
      tokenKey: nextcloud-token
      # smtpUsernameKey: smtp-username
      # smtpPasswordKey: smtp-password
    extraEnv:
      - name: OVERWRITEPROTOCOL
        value: https
      - name: TRUSTED_PROXIES
        value: '0.0.0.0/0'
      - name: PHP_MEMORY_LIMIT
        value: 512M
  nginx:
    enabled: true
  externalDatabase:
    enabled: true
    type: postgresql
    host: nextcloud-postgresql
    database: nextcloud
    existingSecret:
      enabled: true
      secretName: nextcloud.nextcloud-postgresql.credentials.postgresql.acid.zalan.do
      usernameKey: username
      passwordKey: password
    architecture: standalone
  redis:
    enabled: true
    image:
      tag: 6.2.10-debian-11-r14
    auth:
      enabled: true
      password: 'SEE_BITWARDEN'
    replica:
      persistence:
        enabled: false
    master:
      persistence:
        enabled: false
  cronjob:
    enabled: true
    image: 
      repository: curlimages/curl
      tag: 8.5.0
      pullPolicy: IfNotPresent
    schedule: "*/5 * * * *"
    curlInsecure: false
    failedJobsHistoryLimit: 2
    successfulJobsHistoryLimit: 2

  persistence:
    enabled: true
    storageClass: "longhorn"
    accessMode: ReadWriteOnce
    size: 20Gi

```

## Additional context, if any

Missing in nginx config

```
  include mime.types;
  types {
      application/javascript js mjs;
  }
```

### PR DESCRIPTION
**Pull request by Lerentis**
# Pull Request

## Description of the change

added missing MIME type.

## Benefits

Ensure all apps work and remove warning in instance check

## Possible drawbacks

N/A

## Applicable issues

- fixes #519

## Checklist 

- [x] DCO has been [signed off on the commit](https://docs.github.com/en/github/authenticating-to-github/signing-commits).
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).

### CODE REVIEW

**Comment by provokateurin** [comment]
Right, so this is documented in https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html. Please add such documentation links to make it easier to understand why a change was made or where it comes from.
I guess in general we need to make sure those configs are always up-to-date.

File: charts/nextcloud/templates/nginx-config.yaml [hunk 1]
```diff
@@ -14,6 +14,11 @@
         # could take several months.
         #add_header Strict-Transport-Security "max-age=15768000; includeSubDomains; preload;" always;
 
+        include mime.types;
+        types {
+          application/javascript js mjs;
+        }
+
```
Comment (provokateurin) [inline charts/nextcloud/templates/nginx-config.yaml:21]: Please move this down to the same location as it is in https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: charts/nextcloud/templates/nginx-config.yaml (hunk 1 above)
Comment (Lerentis) [reply charts/nextcloud/templates/nginx-config.yaml]: moved :+1: 
there are multiple mismatches i think. `root` is also in a different place than in the docs

Comment (provokateurin) [reply charts/nextcloud/templates/nginx-config.yaml]: Yeah, as I said we need to align them again. It helps a lot though to already put some pieces in the right place.

**Review by provokateurin — APPROVED** [review]

**Pull request merged**